dissect/target/plugins/os/unix/log/messages.py

```diff
@@ -42,39 +51,33 @@ def messages(self):
         Sources:
             - https://geek-university.com/linux/var-log-messages-file/
         """
-
-        log_files: [TargetPath] = list(self.target.fs.path("/var/log/").glob("syslog*")) + list(
-            self.target.fs.path("/var/log/").glob("messages*")
-        )
+        syslogs = list(self.target.fs.path("/var/log/").glob("syslog*"))
+        messages = list(self.target.fs.path("/var/log/").glob("messages*"))
+        log_files: Generator[TargetPath] = chain(syslogs, messages)
```

Comment:
```suggestion
        syslogs = self.target.fs.path("/var/log/").glob("syslog*")
        messages = self.target.fs.path("/var/log/").glob("messages*")
        log_files: Iterator[TargetPath] = chain(syslogs, messages)
```
The idea here is that you chain the generators from `glob`. Right now you completely exhaust the `glob` generators before starting to parse files. By chaining their respective generators you evaluate the glob lazily and can start returning output earlier.